Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -372,63 +370,79 @@ - (void) connectToHostWithName: (NSString*)hostname
               networkInterface: (NSString*)interface {
     CBLLogInfo(WebSocket, @"%@: Connect to host '%@' port '%ld' interface '%@'",
                self, hostname, (long)port, interface);
-    [self resolveAndConnect: hostname port: port networkInterface: interface];
-}
-
-- (void) resolveAndConnect: (NSString*)hostname
-                      port: (NSInteger)port
-          networkInterface: (NSString*)interface {
-    // Get address info:
-    struct addrinfo hints;
-    memset(&hints, 0, sizeof(hints));
-    hints.ai_socktype = SOCK_STREAM;
-    
-    const char* cHost = [hostname cStringUsingEncoding: NSUTF8StringEncoding];
-    const char* cPort = [$sprintf(@"%ld", (long)port) cStringUsingEncoding: NSUTF8StringEncoding];
-    
-    int res = getaddrinfo(cHost, cPort, &hints, &_addr);
-    if (res) {
-        NSString* msg = $sprintf(@"Failed to get address info with error %d", res);
-        CBLWarnError(WebSocket, @"%@: %@", self, msg);
-        [self closeWithError: addrInfoError(res, msg)];
-        return;
-    }
     
-    CBLLogVerbose(WebSocket, @"%@: Host '%@' was resolved as %@", self, hostname, addrInfo(_addr));
-    
-    if (_addr->ai_family != AF_INET && _addr->ai_family != AF_INET6) {
-        NSString* msg = $sprintf(@"Address family %d is not supported", _addr->ai_family);
+    unsigned int index = if_nametoindex([interface cStringUsingEncoding: NSUTF8StringEncoding]);
+    if (index == 0) {
+        int errNo = errno;
+        NSString* msg = $sprintf(@"Failed to find network interface %@ with errno %d", interface, errNo);
         CBLWarnError(WebSocket, @"%@: %@", self, msg);
-        [self closeWithError: posixError(EPERM, msg)];
+        [self closeWithError: posixError(errNo, msg)];
         return;
     }
-    
-    // Create socket:
-    [self _socketConnect: interface];
+
```

Comment:
Move from _socketConnect: to here : 
CBLLogVerbose(WebSocket, @"%@: Interface '%@' is mapped to index '%u'", self, _networkInterface, index);